libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -310,133 +275,64 @@ - (void)enumerateDelegates:(void (^)(id<SFUserAccountManagerDelegate>))block
-}
-
 - (BOOL)isCurrentUserAnonymous {
-    return [[self class] isUserAnonymous:self.currentUser];
-}
-
-- (SFUserAccount *)anonymousUser {
-    if (nil == _anonymousUser) {
-        for (SFUserAccountIdentity *identity in self.allUserIdentities) {
-            if ([identity isEqual:self.anonymousUserIdentity]) {
-                _anonymousUser = [self userAccountForUserIdentity:identity];
-                break;
-            }
-        }
-    }
-    return _anonymousUser;
-}
-
-- (void)setupAnonymousUser:(BOOL)supportsAnonymousUser autocreateAnonymousUser:(BOOL)autocreateAnonymousUser {
-
-    // If there is no current user but the application support anonymous user
-    // and wants it to be created automatically, then create it now.
-    if (supportsAnonymousUser && autocreateAnonymousUser && nil == self.anonymousUser) {
-        [self log:SFLogLevelInfo msg:@"Creating anonymous user"];
-        [self enableAnonymousAccount];
-        if (nil == self.currentUser) {
-            self.currentUser = self.anonymousUser;
-        }
-    }
-}
-
-- (void)enableAnonymousAccount {
-    if (nil == self.anonymousUser) {
-        self.anonymousUser = [[SFUserAccount alloc] initWithIdentifier:[self uniqueUserAccountIdentifier] clientId:self.oauthClientId];
-        SFOAuthCredentials *creds = self.anonymousUser.credentials;
-        creds.domain = self.loginHost;
-        creds.redirectUri = self.oauthCompletionUrl;
-        creds.clientId = self.oauthClientId;
-        creds.identityUrl = [NSURL URLWithString:[NSString stringWithFormat:@"https://does.not.matter.domain-does-not-exist/id/%@/%@", self.anonymousUserIdentity.orgId, self.anonymousUserIdentity.userId]];
-        creds.communityUrl = [NSURL URLWithString:@"https://who-cares.domain-does-not-exist"];
-        creds.instanceUrl = [NSURL URLWithString:@"https://who-cares.domain-does-not-exist"];
-        creds.accessToken = [NSUUID UUID].UUIDString;
-        [self addAccount:self.anonymousUser];
-        [self saveAccounts:nil];
-    }
-}
-
-- (void)disableAnonymousAccount {
-    NSError *error;
-    if (![self deleteAccountForUser:self.anonymousUser error:&error]) {
-        [self log:SFLogLevelError format:@"Unable to delete the anonymous user: %@", [error localizedDescription]];
-    }
-    self.anonymousUser = nil;
+    return self.currentUser == nil;
 }
 
 #pragma mark Account management
-
 - (NSArray *)allUserAccounts
 {
-    // Only load accounts if they haven't yet been loaded.
-    if ([self.userAccountMap count] == 0) {
-        [self loadAccounts:nil];
-    }
-    if ([self.userAccountMap count] == 0) {
-        return nil;
-    }
-    
-    NSMutableArray *accounts = [NSMutableArray array];
-    for (SFUserAccountIdentity *key in [self.userAccountMap allKeys]) {
-        if ([key isEqual:self.temporaryUserIdentity]) {
-            continue;
+    NSMutableArray *accounts = nil;
```

Comment:
I don't think there's any reason for the bulk of this method to exist anymore.  Seems like we could just return `[self.userAccountMap allValues]`.  The special casing doesn't seem necessary.